src/components/Carousel.js

```diff
@@ -443,10 +482,26 @@ class Carousel extends Component {
           position = this.props.infiniteLoop ? 0 : lastPosition;
         }
 
-        this.selectItem({
-            // if it's not a slider, we don't need to set position here
-            selectedItem: position
-        });
+        if (needClonedSlide) {
+            this.selectItem({
+                selectedItem: position,
+                loop: true
+            }, () => {
+                requestAnimationFrame(() => {
+                    requestAnimationFrame(() => {
+                        this.selectItem({
+                            loop: false
```

Comment:
This would break anyone providing onChange handler as we're not passing selectedItem which is referenced inside selectItem.

Reply:
You are right, this could be a setState instead...